### ISSUE TITLE
helm package --sign throws "Error: private key not found" (GPG)

### ISSUE THREAD

**Issue by SimonAlling** [helm/helm#7631]
A confusing error is thrown when signing a chart using a fingerprint to identify a GPG key.

<details>
<summary>Excerpt from provenance docs</summary>

> Once ready to package, add the `--sign` flag to `helm package`. Also, specify
> the name under which the signing key is known and the keyring containing the
> corresponding private key:
> 
> ```
> $ helm package --sign --key 'helm signing key' --keyring path/to/keyring.secret mychart
> ```
> 
> **TIP:** for GnuPG users, your secret keyring is in `~/.gnupg/secring.gpg`. You
> can use `gpg --list-secret-keys` to list the keys you have.
> 
> **Warning:**  the GnuPG v2 store your secret keyring using a new format `kbx` on
> the default location  `~/.gnupg/pubring.kbx`. Please use the following command
> to convert your keyring to the legacy gpg format:
> 
> ```
> $ gpg --export-secret-keys >~/.gnupg/secring.gpg
> ```

</details>

I tried to follow the instructions [here](https://helm.sh/docs/topics/provenance/#the-workflow) (shown above), but ran into this error:

```
$ gpg --version
gpg (GnuPG) 2.2.12
libgcrypt 1.8.4
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/alling/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2

$ gpg --gen-key

$ gpg --export-secret-keys >~/.gnupg/secring.gpg

$ gpg --list-secret-keys 
/home/alling/.gnupg/pubring.kbx
-------------------------------
sec   rsa3072 2020-02-19 [SC] [expires: 2022-02-18]
      4EDF6F4B1CB5B53278AE2CF53AFA909880798BE6
uid           [ultimate] Simon Alling <simon.alling@example.com>
ssb   rsa3072 2020-02-19 [E] [expires: 2022-02-18]

$ helm create mychart
Creating mychart

$ helm package --sign --key '4EDF6F4B1CB5B53278AE2CF53AFA909880798BE6' --keyring ~/.gnupg/secring.gpg mychart
Error: private key not found
```

However, this works:

```
$ helm package --sign --key 'Simon Alling' --keyring ~/.gnupg/secring.gpg mychart
Password for key "Simon Alling <simon.alling@example.com>" >  
Successfully packaged chart and saved it to: /home/alling/mychart-0.1.0.tgz
```

It was far from obvious to me that `helm signing key` was to be interpreted as "your name", though.

**Output of `helm version`:**
```
version.BuildInfo{Version:"v3.0.1", GitCommit:"7c22ef9ce89e0ebeb7125ba2ebf7d421f3e82ffa", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

**Output of `kubectl version`:**
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-12T13:43:46Z", GoVersion:"go1.13.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by bacongobbler** [comment]
fixed in helm/helm-www#500.

**Comment by SimonAlling** [comment]
> fixed in [helm/helm-www#500](https://github.com/helm/helm-www/pull/500).

Depends on your definition of "fixed", I guess. I think it's more intuitive to use the fingerprint; I also believe it's necessary to distinguish between keys with identical `uid`.

**Comment by lindhe** [comment]
AFAICT, the library used does not provide an easy way to check the fingerprint. I hope I'm wrong, but I guess that might be a reason why there's ongoing discussions about changing from that library: #7599
